app/js/lib/foxbox.js

```diff
@@ -327,6 +326,25 @@ export default class Foxbox extends Service {
     });
   }
 
+  performServiceCommand(id, command, method = 'GET', state = undefined) {
```

Comment:
Yeah, I see your point, it sounds like something that deserves a separate PR and a bit of discussion - we'll likely have time in MTV to do this :) Foxbox lib should provide some sort of extension point so that anything like ```getSnapshot``` or ```castMyImage``` won't require changes in the lib and at the same time lib will provide useful functionality for these methods (eg. adding auth headers).

So would be it OK for you to live with ```performServiceCommand``` in this PR + I file and handle the issue to make it more generic and nice as we'll discuss in MTV ? :)